tools/shared/src/main/scala/org/scalajs/core/tools/linker/frontend/optimizer/OptimizerCore.scala

```diff
@@ -3766,7 +3964,7 @@ private[optimizer] object OptimizerCore {
   }
 
   private object PreTransTree {
-    def apply(tree: Tree): PreTransTree = {
+    def make(tree: Tree): PreTransTree = {
```

Comment:
If this deserves to be a different operation (and not inlined into `apply`), it deserves a better name. (And maybe `apply` as well).
Notably: If some of the things done inside `apply` are not correct in some contexts, consider renaming `apply` instead. IMHO calling `A(x, y, z)` should "always work" (more formally: should work if `{x, y, z}` satisfies a *minimal* set of preconditions).